app/models/concerns/noticed/deliverable.rb

```diff
@@ -63,28 +65,34 @@ def deliver(recipients = nil, options = {})
     def deliver(recipients = nil, options = {})
       validate!
 
-      transaction do
-        save!
-
-        recipients_attributes = Array.wrap(recipients).map do |recipient|
-          recipient_attributes_for(recipient)
-        end
-
-        if Rails.gem_version >= Gem::Version.new("7.0.0.alpha1")
-          notifications.insert_all!(recipients_attributes, record_timestamps: true) if recipients_attributes.any?
-        else
-          time = Time.current
-          recipients_attributes.each do |attributes|
-            attributes[:created_at] = time
-            attributes[:updated_at] = time
+      run_callbacks :deliver do
```

Comment:
Upon further thoughts, I am not sure if this should wrap the whole block.  I think it should probably just wrap the `EventJob.set(options).perform_later(self)` call.  What do you think?

Reply:
I think, either way, there might be a risk that the notification as not been delivered before the call back runs.

Reply:
You are correct, but that was how it was implemented in [1.6.3 version](https://github.com/excid3/noticed/blob/v1.6.3/lib/noticed/base.rb#L69) for perform later.  Since everything is performed later, this would be consistent.

If people need something to be guaranteed to be run after a delivery, they would need to use the `DeliveryMethod` callbacks.